src/main/scala/sigmastate/trees.scala

```diff
@@ -104,7 +109,7 @@ case class AND(input: Value[SCollection[SBoolean.type]])
 
   //todo: reduce such boilerplate around AND/OR, folders, map etc
   override def transformationReady: Boolean =
-    input.evaluated && input.items.forall(_.evaluated)
+    input.evaluated && input.fold(_.items.forall(_.evaluated), c => true)
```

Comment:
why are you reducing in transformationReady() ?

Reply:
this is not reducing. This probably confusing function name. It actualy does pattern matching ConcreteColletion and CollectionConstant cases. Check the function.
I can rename it in the next PR.